Gems/PhysX/Code/Tests/CharacterGameplayControllerTests.cpp

```diff
@@ -124,4 +141,35 @@ namespace PhysX
             EXPECT_THAT(expectedGravityMultiplier + i, testing::FloatEq(basis.m_gameplayController->GetGravityMultiplier()));
         }
     }
+
+    TEST_F(PhysXDefaultWorldTest, CharacterGameplayController_EntityFallsUnderGravity)
+    {
+        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight, DefaultFloorTransform);
+
+        // Remove the Ground so we can can test falling without worrying about accidental collisions with objects.
+        basis.DeleteGround();        
+        
+        // Let scene run for a few moments so the entity can be manipulated by gravity from the gameplay component
+        const auto startPosition = basis.m_controller->GetPosition();
+        const int timeStepCount = rand() % 180 + 10;
```

Comment:
Please avoid using random values in the test. The unit tests need to be deterministic and use the same values to determine if when the code changes will still pass the test or not. Adding randomness eliminates that certainty since different executions will use different values, which potentially can pass on some values and fail in another.

If testing with different amount of time steps is relevant then I would recommend choosing a deliberate amount of values and always test them all. You can use the following approach to run the same test with different values:

````
// Use the PhysXDefaultWorldTestWithParam class for the fixture to indicate the parameter type is going to be int
using PhysXDefaultWorldTestWithParamFixture= PhysXDefaultWorldTestWithParam<int>;

// Use the new PhysXDefaultWorldTestWithParamFixture fixture.
TEST_F(PhysXDefaultWorldTestWithParamFixture, CharacterGameplayController_EntityFallsUnderGravity)
    {
        GameplayTestBasis basis(m_testSceneHandle, DefaultGravityMultiplier, DefaultGroundDetectionBoxHeight, DefaultFloorTransform);

        // Remove the Ground so we can can test falling without worrying about accidental collisions with objects.
        basis.DeleteGround();        

        // Let scene run for a few moments so the entity can be manipulated by gravity from the gameplay component
        const auto startPosition = basis.m_controller->GetPosition();
        const int timeStepCount = GetParam(); // <-- Gets the current value to use.

       ...
}

// Indicate the values to call all the tests that use PhysXDefaultWorldTestWithParamFixture.
INSTANTIATE_TEST_CASE_P(PhysXDefaultWorldTest, PhysXDefaultWorldTestWithParamFixture, ::testing::Values(10, 30, 60, 90, 120, 136, 190));
````

Reply:
+1 to @moraaar's comment, though one consideration is it might be okay to use random so long as you use a fixed seed (so the same random numbers are produced each run). If you do want to do this it's also best to avoid `rand` and use `uniform_int_distribution` and a `random_device` (see https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution for more info)

Reply:
Thanks for this. After previous discussions about the relative error in the distance calculations, it became clear that a few different times should be used.

I'm going to use the approach that @moraaar has provided to strengthen this test.